exercises/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+type Entry struct {
+	ID          uint32
+	FirstName   string
+	LastName    string
+	PhoneNumber string
+}
+
+func newEntry() {
+	const phonebookFile = "phonebook.json"
+	var entries []Entry
+	var ID uint32
+	e := Entry{ID: ID}
+	f, err := os.Open(phonebookFile)
```

Comment:
I recommend doing reading in a separate function to avoid dups
```go
func loadPhonebook() (e []Entries, err error) {
  f, err := os.Open(phonebookFile)
  if err != nil {
    return nil, err
  }
  defer f.Close()
  err = json.NewDecoder(f).Decode(&e)
  return e, err
}

func addEntry() {
  entries, err := loadPhonebook()
  ...
  maxID := maxEntryID(entries)
  e, err := newEntryFromUser(maxID+1)
  entries = append(entries, e)
  err = saveEntries(entries)
  ...
}

func listEntry() {
  entries, err := loadPhonebook()
  ...
}
```